buildSrc/src/main/kotlin/kotest-native-conventions.gradle.kts

```diff
@@ -20,13 +20,11 @@ kotlin {
 
       watchosArm32()
       watchosArm64()
-      watchosX86()
       watchosX64()
       watchosSimulatorArm64()
 
       iosX64()
       iosArm64()
-      iosArm32()
       iosSimulatorArm64()
 
       sourceSets {
```

Comment:
Could we just replace the entire `sourceSets` block with `applyDefaultHierarchyTemplate()`? I'd say the [full hierarchy template﻿](https://kotlinlang.org/docs/multiplatform-hierarchy.html#see-the-full-hierarchy-template) has got us covered and the intermediate `desktopMain` is no longer necessary (will be covered by `nativeMain` automatically).